server/src/test/java/io/crate/metadata/upgrade/IndexTemplateUpgraderTest.java

```diff
@@ -362,4 +365,80 @@ public void test_populateColumnPositions_method_fixes_duplicates() {
+
+        var result = IndexTemplateUpgrader.fixCorruptedTemplatesFromSwapTableBug(Map.of(partnerName, partner, dummyName, dummy, corruptedName, corrupted));
+
+        IndexTemplateMetadata indexTemplateMetadata = result.get(partnerName);
+        assertThat(indexTemplateMetadata).isNotNull();
+        assertThat(indexTemplateMetadata.patterns().size()).isEqualTo(1);
+        assertThat(indexTemplateMetadata.patterns().get(0)).isEqualTo(partnerName);
+        // mappings and settings are swapped
+        assertThat(indexTemplateMetadata.mappings().get(DEFAULT_MAPPING_TYPE).toString()).isEqualTo(mappingWithinCorruptedTemplate);
+        assertThat(indexTemplateMetadata.settings().get(INDEX_REFRESH_INTERVAL_SETTING.getKey())).isEqualTo("300ms");
+
+        // name is fixed, _broken appended
+        String brokenName = "m1..partitioned.s1_broken.";
+        IndexTemplateMetadata broken = result.get(brokenName);
+        assertThat(broken).isNotNull();
+        assertThat(broken.patterns().size()).isEqualTo(1);
+        assertThat(broken.patterns().get(0)).isEqualTo(brokenName + "*");
+        // mappings and settings are swapped
+        assertThat(broken.mappings().get(DEFAULT_MAPPING_TYPE).toString()).isEqualTo(mappingWithinPartnerTemplate);
+        assertThat(broken.settings().get(INDEX_REFRESH_INTERVAL_SETTING.getKey())).isEqualTo("400ms");
+
+        //dummy is untouched
+        IndexTemplateMetadata dummyTemplate = result.get(dummyName);
+        assertThat(dummyTemplate).isNotNull();
+        assertThat(dummyTemplate.patterns().size()).isEqualTo(1);
+        assertThat(dummyTemplate.patterns().get(0)).isEqualTo(dummyName);
+        assertThat(dummyTemplate.mappings().get(DEFAULT_MAPPING_TYPE).toString()).isEqualTo(dummyMapping);
+        assertThat(dummyTemplate.settings().get(INDEX_REFRESH_INTERVAL_SETTING.getKey())).isEqualTo("500ms");
+    }
+
+    @Test
+    public void test_method_fixCorruptedTemplatesFromSwapTableBug_corrupted_template_without_swap_table_partner() throws IOException {
+        String corruptedName = ".partitioned.m1.s1.";
+        String mappingWithinCorruptedTemplate = "{\"default\":{\"_meta\":{\"partitioned_by\":[[\"t\",\"boolean\"]]},\"dynamic\":\"strict\",\"properties\":{\"t\":{\"index\":false,\"position\":1,\"type\":\"boolean\"}}}}";
+        IndexTemplateMetadata.Builder corruptedBuilder = new IndexTemplateMetadata.Builder(corruptedName);
+        corruptedBuilder.patterns(List.of(corruptedName));
+        corruptedBuilder.settings(Settings.builder().put(INDEX_REFRESH_INTERVAL_SETTING.getKey(), new TimeValue(300)));
+        corruptedBuilder.putMapping(DEFAULT_MAPPING_TYPE, mappingWithinCorruptedTemplate);
+        IndexTemplateMetadata corrupted = corruptedBuilder.build();
+
+        var result = IndexTemplateUpgrader.fixCorruptedTemplatesFromSwapTableBug(Map.of(corruptedName, corrupted));
+
+        String fixedCorruptedName = "m1..partitioned.s1.";
```

Comment:
Afaik a valid template name in our case won't end with `.`.

Reply:
I think the naming should be consistent to `PartitionName.templateName` which appends `.` at the end.